src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -33,6 +33,49 @@ void align_u4_zp_constants(const std::shared_ptr<ov::Model>& model) {
     }
 }
 
+// FIXME: It should be different function or at least this one requires fix
+uint32_t calculate_trainable_parameters(const std::shared_ptr<ov::Model>& model) {
+    uint32_t total_parameters = 0u;
+    for (auto op : model->get_ops()) {
+        for (auto input : op->inputs()) {
+            auto src = input.get_source_output().get_node_shared_ptr();
+            if (auto cst_op = std::dynamic_pointer_cast<ov::op::v0::Constant>(src)) {
+                total_parameters += ov::shape_size(cst_op->get_shape());
+            }
+        }
+    }
+    std::cout << "total params: " << static_cast<float>(total_parameters) / 1'000'000'000 << std::endl;
+    return total_parameters;
+}
+
+bool allow_to_enable_npuw_dq(const std::shared_ptr<ov::Model>& model, const bool is_prefill_model = false) {
+    std::vector<std::string> rt_info_path = {"nncf", "weight_compression", "group_size"};
+    if (!model->has_rt_info(rt_info_path)) {
+        // NB: Model isn't compressed by using either CW or GQ configurations
+        return false;
+    }
+    auto group_size = model->get_rt_info<int>(rt_info_path);
+    if (group_size == -1) {
+        // NB: For CW configuration NPUW DQ is enabled unconditionally
+        return true;
+    }
+    const uint32_t kTotalParams = calculate_trainable_parameters(model);
+    if (!is_prefill_model && kTotalParams < 2'000'000'000) {
+        // NB: For GQ configurations NPUW DQ is enabled only
+        // for models that have <2B parameters and only for generate model (so far)
+        return true;
+    }
+    return false;
+}
+
+void enable_npuw_dq_if_allowed(ov::AnyMap& config,
+                               const std::shared_ptr<ov::Model>& model,
+                               const bool is_prefill_model = false) {
+    if (allow_to_enable_npuw_dq(model, is_prefill_model)) {
+        config["NPUW_DQ"] = "YES";
```

Comment:
OpenVINO already provides an option for dynamic quantization which is called "DYNAMIC_QUANTIZATION_GROUP_SIZE". Details are here: https://docs.openvino.ai/2024/learn-openvino/llm_inference_guide/llm-inference-hf.html#enabling-openvino-runtime-optimizations.

It would be great to have an alignment here.

Reply:
I believe there's some mess in the terminology used by various projects, let's assume ours mean "DeQuantization" - when it is done with some help from the plugin

Reply:
Nice catch, fixed

Reply:
Reverted :)